src/app/functions/server/onerep.ts

```diff
@@ -265,6 +266,16 @@ export async function isEligible() {
     throw new Error("No session");
   }
 
+  const flag = await getFlag("FreeBrokerScan");
+
+  if (!flag.isEnabled) {
+    throw new Error("Flag is not enabled");
+  }
+
+  if (!flag.allowList?.includes(session.user.email)) {
```

Comment:
Not sure if it matters, but I have cursed eyes. Elsewhere in these files we're doing:
```js
  if (!session?.user?.email) {
    return redirect("/");
  }
```

Does this need the additional optional chaining on `session?.user?.email`? Or maybe not if we're aborting earlier on L265-267 (although that's checking `session.user.subscriber.id`, but one might infer the other):
```js
  if (!session?.user?.subscriber?.id) {
    throw new Error("No session");
  }
```